### PR TITLE
#105: Introducing Raid command and sub-commands

### LINKED ISSUE #104: Command[Raid]: Investigate possibility of a raid command
Command Usage: 
?raid <pokemon> <gym> <hatch_time>
?raid <pokemon> <gym>
?raid <pokemon> train

Has to follow one of three formats: [?raid heatran klockstapeln 18:15], [?raid heatran train], [?raid heatran klockstapeln].

Creates a new raid channel under the same category called pokemon_gym (#heatran_klockstapeln_1815, #heatran_klockstapeln, #heatran_train).
Embed footer says, info can be updated with ?update [gym, time, pokemon].
If pokemon can't be found in dex title says pokemon name, description says exact pokemon couldn't be found in dex, ?update pokemon.

Delete channel by ?raid close. Sneasel reminds every 1 hour with a message to ?raid close and mentions reporter, scheduled task is removed with ?raid close.

### LINKED ISSUE #105: Command[Raid]: Raid feature to create individual channels
From Investigation: 

Command Usage:
?raid <proposed_time>
?raid train

Has to follow one of two formats: [?raid heatran klockstapeln 18:15], [?raid heatran train].

Creates a new raid channel under the same category called pokemon_gym (#heatran_klockstapeln, #heatran_train).
If channel already exists with that name then the channel is called pokemon_position_2 (incrementing).
Embed footer says, info can be updated with ?update [gym, time, pokemon].
If pokemon can't be found in dex title says pokemon name, description says exact pokemon couldn't be found in dex, ?update pokemon.

Delete channel by ?raid close. Sneasel reminds every 1 hour with a message to ?raid close and mentions reporter, scheduled task is removed with ?raid close.

### LINKED ISSUE #106: Command[Raid]: Update command
Add ?update command group with sub-commands pokemon, gym, time.
Should verify that the channel is a raid channel, should only be callable by reporter and admin/mods.
Has to update the embed in channel and channel_name.

### LINKED ISSUE #107: Command[Raid]: Close feature
Implement ?close to close opened raid channels.
Should verify that it is a raid channel and should only be callable by reporter and admin/mods.
Upon close should also remove scheduled event with correct tag: sendCHANNEL_ID

### LINKED ISSUE #108: Command[Dex]: Enable mega evolutions once data available

### LINKED ISSUE #109: Command[Raid]: Add support for spelling errors
Could also be useful for dex

### LINKED ISSUE #111: Command[Raid]: Add status command that reposts the raid's embed
Usage: ?status

### LINKED ISSUE #114: Command[Raids]: List all active raids
?raids to list all active raids, get list of active raids from database and mention all channels through the channel_id in each row.

Also remore remoteraids from raid reporting channel.
Which channels should ?raids be allowed in?

### PR DESCRIPTION
**Pull request by NickWedzinga**
Closes #104, closes #109, closes #114, closes #108, closes #106, closes #107, closes #111 and closes #105

### CODE REVIEW

**Review by Schanihbg — APPROVED** [review]

File: sneaselcommands/raids/raid.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    await ctx.send(f"{ctx.author.mention} has created a raid in {created_channel.mention}, why don't you join them?")
+    await _send_embed(bot, ctx, created_channel, pokemon, gym, hatch_time)
+
+
+class Raid(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    # TODO: rename table configure__schedule to configure__schedule_weekly
+    # TODO: create separate categories for each city for raid reporting, rename karlskronaraids and other city raids to "raids" or something
```
Comment (NickWedzinga) [inline sneaselcommands/raids/raid.py:218]: remove before merge

File: sneaselcommands/raids/raids.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+
+def _create_embed(ctx, active_raids_dict: dict, max_field_count: int = 24):
+    """Creates an embed listing all active raids, max of 24 listings"""
+    embed = discord.Embed(title=f"Active raids",
+                          color=0xff9900,
+                          description="Visit the raid channels below to join the discussion",
+                          timestamp=datetime.utcnow())
+
+    if len(active_raids_dict) == 0:
```
Comment (NickWedzinga) [inline sneaselcommands/raids/raids.py:21]: if active_raids_dict, redundant len check

File: testing/integration/integration_manager.py [hunk 3]
```diff
@@ -41,18 +42,19 @@ async def test(self, ctx):
...
             await ctx.send(
                 f"Sending results to {common.TEST_RESULTS_CHANNEL.mention}. This may take a while :sweat_smile:")
 
             # TODO: Add all the tests to a list or something to run
-            await leaderboard_integration.run_tests(self.bot, ctx)
-            await list_integration.run_tests(ctx, self.bot)
-            await support_integration.run_tests(ctx, self.bot)
-            await configure_integration.run_tests(self.bot, ctx)
-            await dex_integration.run_tests(self.bot, ctx)
+            # await leaderboard_integration.run_tests(self.bot, ctx)
```
Comment (NickWedzinga) [inline testing/integration/integration_manager.py:52]: oops

File: utils/channel_wrapper.py [hunk 4]
```diff
@@ -36,3 +37,20 @@ async def delete_channel(bot, channel, source: str = None):
...
+
+
+async def find_embed_in_channel(bot, channel: Optional[discord.TextChannel], source: str, should_pm: bool = True) -> Optional[discord.Message]:
+    """
+    Returns the first message containing an embed in the provided channel
+    If the channel/embed doesn't exist, will pm_dev_error and return None
+    """
+    if channel is not None:
+        async for message in channel.history(oldest_first=True):
+            if len(message.embeds) > 0:
```
Comment (NickWedzinga) [inline utils/channel_wrapper.py:49]: if message.embeds

**Pull request merged**